### PR TITLE
Clean up FsRepositoryTests

### PR DESCRIPTION
**Pull request by pxsalehi**
I might be mistaken, but the CountDownLatch doesn't seem to be necessary.

### CODE REVIEW

File: server/src/test/java/org/elasticsearch/repositories/fs/FsRepositoryTests.java [hunk 1]
```diff
@@ -201,16 +200,8 @@ public void testSnapshotAndRestore() throws IOException, InterruptedException {
...
-        CountDownLatch latch = new CountDownLatch(1);
-        threadPool.generic().submit(() -> {
-            try {
-                runnable.run();
-            } finally {
-                latch.countDown();
-            }
-        });
-        latch.await();
+    private void runGeneric(ThreadPool threadPool, Runnable runnable) {
```
Comment (DaveCTurner) [inline server/src/test/java/org/elasticsearch/repositories/fs/FsRepositoryTests.java:203]: may as well just inline this now it's a one-liner?

**Review by DaveCTurner — CHANGES_REQUESTED** [review]
Yep I think you're right. These APIs weren't always completely async, but now that it is we can do this. But I think we can go further: there's no need to run this stuff on a background thread  at all now, and we can even assert that each future is complete when the corresponding method returns.

File: server/src/test/java/org/elasticsearch/repositories/fs/FsRepositoryTests.java (hunk 1 above)
Comment (DaveCTurner) [reply server/src/test/java/org/elasticsearch/repositories/fs/FsRepositoryTests.java:203]: ah sorry meant to delete this comment - there's no need to fork at all any more I think

Comment (pxsalehi) [reply server/src/test/java/org/elasticsearch/repositories/fs/FsRepositoryTests.java]: :facepalm: right! I've also done some renames! Please see https://github.com/elastic/elasticsearch/pull/88723/commits/d942ffcf9e51dd0a2e8ca068cd1c5abd81286483.

**Comment by DaveCTurner** [comment]
> That means we're never actually running any data file uploads or restores that would fork wouldn't it?

Not sure about all such tests but at least here we're using the mock threadpool that `BlobStoreTestUtil#mockClusterService` creates which runs everything using a `SameThreadExecutorService`.

**Comment by original-brownbear** [comment]
> Not sure about all such tests but at least here we're using the mock threadpool that BlobStoreTestUtil#mockClusterService creates which runs everything using a SameThreadExecutorService.

Ah right the test threadpool is now unused ... my bad ++ to asserting that things are completed right away then.

**Review by DaveCTurner — COMMENTED** [review]
> Ah right the test threadpool is now unused

Good point yes `ThreadPool threadPool = new TestThreadPool(getClass().getSimpleName());` is no longer needed here.

**Comment by DaveCTurner** [comment]
image

I'm also a big fan of putting the date in my branch names :) You might find it useful to use a fully numeric YYYY-MM-DD rather than the month name - it sorts better.

**Comment by pxsalehi** [comment]
Yay! Even more code removed! :-)
Please see https://github.com/elastic/elasticsearch/pull/88723/commits/c112c3d3b09a3fc66a381c91fcd4bc0689c85905.

**Review by DaveCTurner — APPROVED** [review]
LGTM

**Comment by pxsalehi** [comment]
Thanks David and Armin!

**Pull request merged**